xbmc/guilib/GUITextBox.cpp

```diff
@@ -241,10 +241,22 @@ void CGUITextBox::Render()
 
       while (posY < m_posY + m_renderHeight && current < (int)m_lines.size())
       {
+        const CGUIString& lineString = m_lines[current];
+        float linePosX = posX;
         uint32_t align = alignment;
-        if (m_lines[current].m_text.size() && m_lines[current].m_carriageReturn)
+
+        if (lineString.m_text.size() && lineString.m_carriageReturn)
           align &= ~XBFONT_JUSTIFIED; // last line of a paragraph shouldn't be justified
-        m_font->DrawText(posX, posY, m_colors, m_label.shadowColor, m_lines[current].m_text, align, m_width);
+
+        if (align & XBFONT_RIGHT)
+        {
+          // We need to adjust the posX in similar way the CGUILabel recalculate the render rect
+          // see CGUILabel::UpdateRenderRect()
+          linePosX -= GetTextWidth(lineString.GetAsString());
```

Comment:
That doesn't seem right for unicode? Seems that a CGUIString is a vector of uint32_t, so most likely meant to accept UTF32, to be confirmed though.
A new GetTextWidth that accepts a vecText would preserve unicode data & avoid a trip through string?